cluster-autoscaler/cloudprovider/clusterapi/clusterapi_utils_test.go

```diff
@@ -400,3 +401,202 @@ func TestUtilNormalizedProviderID(t *testing.T) {
 		})
 	}
 }
+
+func TestScaleFromZeroEnabled(t *testing.T) {
+	for _, tc := range []struct {
+		description string
+		enabled     bool
+		annotations map[string]string
+	}{{
+		description: "nil annotations",
+		enabled:     false,
+	}, {
+		description: "empty annotations",
+		annotations: map[string]string{},
+		enabled:     false,
+	}, {
+		description: "non-matching annotation",
+		annotations: map[string]string{"foo": "bar"},
+		enabled:     false,
+	}, {
+		description: "matching key, incomplete annotations",
+		annotations: map[string]string{
+			"foo":  "bar",
+			cpuKey: "1",
+			gpuKey: "2",
+		},
+		enabled: false,
+	}, {
+		description: "matching key, complete annotations",
+		annotations: map[string]string{
+			"foo":     "bar",
+			cpuKey:    "1",
+			memoryKey: "2",
+		},
+		enabled: true,
+	}} {
+		t.Run(tc.description, func(t *testing.T) {
+			got := scaleFromZeroEnabled(tc.annotations)
+			if tc.enabled != got {
+				t.Errorf("expected %t, got %t", tc.enabled, got)
+			}
+		})
+	}
+}
+
+func TestParseCPUCapacity(t *testing.T) {
+	for _, tc := range []struct {
+		description      string
+		annotations      map[string]string
+		expectedQuantity resource.Quantity
+		expectedError    bool
+	}{{
+		description:      "nil annotations",
+		expectedQuantity: zeroQuantity.DeepCopy(),
+		expectedError:    false,
+	}, {
+		description:      "empty annotations",
+		annotations:      map[string]string{},
+		expectedQuantity: zeroQuantity.DeepCopy(),
+		expectedError:    false,
+	}, {
+		description:      "bad quantity",
+		annotations:      map[string]string{cpuKey: "not-a-quantity"},
+		expectedQuantity: zeroQuantity.DeepCopy(),
+		expectedError:    true,
+	}, {
+		description:      "valid quantity",
+		annotations:      map[string]string{cpuKey: "123m"},
+		expectedError:    false,
+		expectedQuantity: resource.MustParse("123m"),
+	}} {
```

Comment:
Should also add quantities that is not in this format, CPU can also be specified like `1`, `2`, `0.1`, and would be good to compare those results as well